packages/database/src/core/SyncTree.ts

```diff
@@ -505,6 +506,42 @@ export class SyncTree {
     );
   }
 
+  getServerValue(query: Query): Node | null {
+    const path = query.path;
+    let serverCache: Node | null = null;
+    let foundAncestorDefaultView = false;
+    // Any covering writes will necessarily be at the root, so really all we need to find is the server cache.
+    // Consider optimizing this once there's a better understanding of what actual behavior will be.
+    this.syncPointTree_.foreachOnPath(path, (pathToSyncPoint, sp) => {
+      const relativePath = Path.relativePath(pathToSyncPoint, path);
+      serverCache = serverCache || sp.getCompleteServerCache(relativePath);
+      foundAncestorDefaultView =
+        foundAncestorDefaultView || sp.hasCompleteView();
+    });
+    let syncPoint = this.syncPointTree_.get(path);
+    if (!syncPoint) {
+      syncPoint = new SyncPoint();
+      this.syncPointTree_ = this.syncPointTree_.set(path, syncPoint);
+    } else {
+      foundAncestorDefaultView =
+        foundAncestorDefaultView || syncPoint.hasCompleteView();
+      serverCache = serverCache || syncPoint.getCompleteServerCache(Path.Empty);
+    }
+    if (serverCache != null) {
+      let serverCacheNode: CacheNode | null = new CacheNode(
+        serverCache,
+        true,
+        false
+      );
+      let writesCache: WriteTreeRef | null = this.pendingWriteTree_.childWrites(
+        query.path
+      );
+      let view: View = syncPoint.getView(query, writesCache, serverCache, true);
+      return view.getCompleteServerCache(query.path);
```

Comment:
@schmidt-sebastian This seems to apply the filters for startAfter/endBefore, but it erroneously returns null for this test: https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/test/query.test.ts#L3159-L3179.

Can you have a look at this? I'm trying to figure out why this is happening but not making much progress.